draft-ietf-lake-edhoc.md

```diff
@@ -462,18 +462,30 @@ Cryptographically, EDHOC does not put requirements on the lower layers. EDHOC is
-* message correlation.
+* message correlation, see {{ci-edhoc}}.
 
-The use of transport is specified in the application profile {{applicability}}.
+EDHOC does not require error free transport since changes in the content of a message is detected by the integrity verification, although possibly in the following message, see "transcript hashes" in {{asym}}.
 
-### Use of Connection Identifiers for EDHOC Message Correlation {#ci-edhoc}
+EDHOC is designed to enable an authenticated key exchange with small messages, where the minimum message sizes are of the order illustrated in the first column of {{fig-sizes}}. There is no maximum message size specified by the protocol; this is for example dependent on the size of authentication credentials (if they are transported, see {{auth-key-id}}).
 
-The transport needs to support the correlation between EDHOC messages and facilitate the retrieval of protocol state and security context during an EDHOC session, including an indication of a message being message_1. The correlation may reuse existing mechanisms in the transport protocol. For example, the CoAP Token may be used to correlate EDHOC messages in a CoAP response and an associated CoAP request.
+The use of transport is specified in the application profile, which in particular may specify limitations in message sizes, see {{applicability}}.
 
-Connection identifiers may be used to correlate EDHOC messages and facilitate the retrieval of protocol state/security context during an EDHOC session. Transports that do not inherently provide correlation across all messages of an EDHOC session can send connection identifiers along with EDHOC messages to gain that required capability, e.g., by prepending the appropriate connection identifier (when available from the EDHOC protocol) to the EDHOC message. Transport of EDHOC in CoAP payloads is described in {{coap}}, which also shows how to use connection identifiers and message_1 indication with CoAP.
+
+### EDHOC Message Correlation {#ci-edhoc}
+
+Correlation between EDHOC messages is needed to facilitate the retrieval of protocol state and security context during an EDHOC session. It is also helpful for the Responder to get an indication that a received EDHOC message is the beginning of a new session so there is no protocol state or security context to retrieve.
+
+Correlation may be based on existing mechanisms in the transport protocol, for example, the CoAP Token may be used to correlate EDHOC messages in a CoAP response and in an associated CoAP request. The connection identifiers may also be used to correlate EDHOC messages.
+
+If correlation between consecutive messages is not provided by other means then the transport SHOULD prepend the appropriate connection identifier (when available from the EDHOC protocol) to the EDHOC message. If message_1 indication is not provided by other means, then the transport SHOULD prepend the message_1 with the CBOR simple value `true` (0xf5), as a unique dummy identifier.
```

Comment:
The SHOULD (instead of SHALL) was written with the knowledge of one specific EDHOC implementation which enables correlation without consideration of transport, simply by only allowing one instance of the protocol at a time. So it seems there is a choice here whether this is solved by "transport" or not. An alternative wording is that we consider dealing with correlation to be an aspect of transport, in which case we could change to SHALL. 

(I don't know if this addresses the comment or even makes sense. I wait with the second comment until we have resolved this.)

Reply:
That's fine for an EDHOC implementation (in fact I'd recommend all this to be handled not by the EDHOC implementation but by the implementation of the transport, eg. EDHOC-over-CoAP).

That implementation is also fine with the proposed changed wording; users of that EDHOC implementation would just have the choice of either not using it with transports that need prepended identifiers (such as over CoAP), or it would need to rely on the EDHOC-over-CoAP part of the surrounding implementation to handle the identifiers.

But nowhere should an implementation that (as a whole) uses prefix-less messages implement the .well-known/edhoc, or any other rt=core.edhoc interface (because it's not compatible with others).